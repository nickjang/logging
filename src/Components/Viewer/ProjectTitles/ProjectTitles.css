.project-titles .selector-title {
  padding-left: 33px
}

.project-title-instructions {
  width: 80%;
  margin: 1rem auto;
  text-align: center;
}