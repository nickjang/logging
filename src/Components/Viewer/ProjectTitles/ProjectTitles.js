import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ProjectTitles.css';

class ProjectTitles extends Component {
  /**
   * Generate titles for each selected project 
   * and it's selectors, if it has any.
   */
  generateProjectTitles = (selectors, projects) => {
    return selectors.map(project => {
      const titles = project.selectors.map(selector =>
        <span key={`selector-${selector.id}`} className='selector-title'>
          {selector.calendar.value + 
          (selector.endRange.value ? `${selector.endRange.value}` : '')}
        </span>
      );

      return (
        <React.Fragment key={project.projectId}>
          <span className='project-title'>{`+ ${projects[project.projectId].name}`}</span>
          {titles}
        </React.Fragment>
      );
    });
  }

  render() {
    const titles = this.generateProjectTitles(this.props.selectors, this.props.projects);

    return (
      <h2 className='project-titles group-col'>
        {titles}
      </h2>
    );
  }
}

ProjectTitles.defaultProps = {
  selectors: [],
  projects: {}
}

ProjectTitles.propTypes = {
  selectors: PropTypes.arrayOf(PropTypes.object).isRequired,
  projects: PropTypes.object.isRequired
}

export default ProjectTitles;