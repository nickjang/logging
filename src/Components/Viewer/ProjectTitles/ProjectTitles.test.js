import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProjectTitles from './ProjectTitles';

describe(`ProjectTitles component`, () => {
  it('renders ProjectTitles by default', () => {
    const wrapper = shallow(<ProjectTitles />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})