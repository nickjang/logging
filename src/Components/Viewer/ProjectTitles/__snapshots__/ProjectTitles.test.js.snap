// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTitles component renders ProjectTitles by default 1`] = `
<p
  className="project-title-instructions note"
>
  To view logs, open the side bar and choose where/when to get logs from
</p>
`;
